### PR TITLE
Implement L0-fetcher

### LINKED ISSUE #1: Create ftp-s3 sync (1)
*Placeholder*

get files from ftp puts in s3

### PR DESCRIPTION
**Pull request by noreklint**
This PR adds L0-fetcher lambda, whichs runs rclone to sync data from SFTP to AWS S3.

Rclone is added as a lambda layer, which is done by first building rclone locally then pushing the zipped dist as part of the fetcher stack.

Resolves #1

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,41 @@
+[tox]
+envlist = py310,mypy,lint
```
Comment (skymandr) [inline tox.ini]: CDK only supports 3.9.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: "3.10"
+
+        - name: Install General Dependencies
+          run: |
+            python3.10 -m pip install --upgrade pip
```
Comment (skymandr) [inline .github/workflows/ci.yml]: See comment on tox.ini

File: fetcher/handlers/l0_fetcher.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+BotoClient = Any
+Event = Dict[str, Any]
+
+FTP_PATH = "/pub/OPS/TM/"
+
+
+def get_rclone_config_path(
+    ssm_client: BotoClient,
```
Comment (skymandr) [inline fetcher/handlers/l0_fetcher.py:16]: You didn't manage to find the client types either, I take it?

File: fetcher/handlers/l0_fetcher.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        f"S3:{bucket}",
+    ]
+
+    if not full_sync:
+        cmd += ["--update", "--use-server-modtime", "--max-age", "7d"]
+
+    return cmd
+
+
+def lambda_handler(event: Event, _):
```
Comment (skymandr) [inline fetcher/handlers/l0_fetcher.py:46]: are you sure that the second argument doesn't have to be named `context`?

**Review by skymandr — APPROVED** [review]
LGTM, though see comments, in particular regarding python version.

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+whitelist_externals =
+    make
+basepython = python3.10
+deps =
+    -rrequirements.txt
+    mypy
+commands =
+    mypy --check-untyped-defs --warn-unused-ignores --no-incremental \
+    	--show-error-codes \
+    	fetcher tests
```
Comment (skymandr) [inline tox.ini:26]: borde inte handlers ingå här förresten?

File: cdk.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      "**"
+    ],
+    "exclude": [
+      "README.md",
+      "cdk*.json",
+      "requirements*.in",
+      "requirements*.txt",
+      "**/__init__.py",
+      "python/__pycache__",
+      "tests"
```
Comment (skymandr) [inline cdk.json]: tox.ini etc?

File: tox.ini (hunk 5 above)
Comment (skymandr) [reply tox.ini:26]: ah, nej. ser att du har ett lite annat katalogupplägg än jag (jag har handler i roten). Ditt är nog bra, men borde inte cdk.json uppdateras i så fall, så att den inte packar med stack-koden?

File: tests/fetcher/test_l0_fetcher_stack.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (skymandr) [inline tests/fetcher/test_l0_fetcher_stack.py:1]:
```suggestion
```

File: fetcher/handlers/l0_fetcher.py (hunk 4 above)
Comment (noreklint) [reply fetcher/handlers/l0_fetcher.py:46]: I think so, I've tried deploying and testing the lambda and no problems detected

File: fetcher/handlers/l0_fetcher.py (hunk 3 above)
Comment (noreklint) [reply fetcher/handlers/l0_fetcher.py:16]: Nope not for the `boto.client` function. Other than that boto seems typed

**Pull request merged**